packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -104,22 +105,193 @@ export default class VTable {
+                        }
+
+                        if (
+                            this.cells.indexOf(row) !== this.cells.length - 1 &&
+                            this.cells.indexOf(row) !== 0
+                        ) {
+                            cell.td.style.borderBottomColor = TRANSPARENT;
+                        }
+                        if (row.indexOf(cell) !== 0) {
+                            cell.td.style.borderLeftColor = TRANSPARENT;
+                        }
+                        if (
+                            row.indexOf(cell) !== this.cells[0].length - 1 &&
+                            row.indexOf(cell) !== 0
+                        ) {
+                            cell.td.style.borderRightColor = TRANSPARENT;
+                        }
+                        if (cell === this.cells[0][0]) {
+                            cell.td.style.borderRightColor = TRANSPARENT;
+                        }
+                    });
+                });
+                break;
+            case TableBorderFormat.removeHeaderRowMiddleBorder:
+                this.forEachCellOfRow(0, (cell, i) => {
+                    if (i !== 0) {
+                        cell.td.style.borderLeftColor = TRANSPARENT;
+                    }
+                    if (i !== this.cells[0].length - 1) {
+                        cell.td.style.borderRightColor = TRANSPARENT;
+                    }
+                });
+                break;
+            case TableBorderFormat.onlyMiddleBorders:
+                this.forEachCellOfColumn(0, (cell, row, i) => {
+                    if (cell.td) {
+                        cell.td.style.borderLeftColor = TRANSPARENT;
+                    }
+                });
+                this.forEachCellOfColumn(this.cells[0].length - 1, (cell, row, i) => {
+                    if (cell.td) {
+                        cell.td.style.borderRightColor = TRANSPARENT;
+                    }
+                });
+        }
+    }
+
+    /**
+     * Apply custom design to the first table column
+     * @param format
+     * @returns
+     */
+    private setFirstColumnFormat(format: Partial<TableFormat>) {
+        if (!format.firstColumn) {
+            return;
+        }
+
+        this.forEachCellOfColumn(0, (cell, row, i) => {
+            if (cell.td) {
+                if (i !== 0) {
+                    cell.td.style.borderTopColor = TRANSPARENT;
+                    cell.td.style.backgroundColor = TRANSPARENT;
+                }
+                if (i !== this.cells.length - 1 && i !== 0) {
+                    cell.td.style.borderBottomColor = TRANSPARENT;
+                }
+            }
+        });
+    }
+
+    /**
+     * Apply custom design to the Header Row
+     * @param format
+     * @returns
+     */
+    private setHeaderRowFormat(format: Partial<TableFormat>) {
+        if (!format.headerRow) {
+            format.headerRowColor = format.bgColor;

```

Comment:
If no header row, you should just return and don't apply any styles or you would override the first column style

Example:
![image](https://user-images.githubusercontent.com/8291124/147292030-de98dc98-0164-4812-9a1d-38ca2b26cc0f.png)